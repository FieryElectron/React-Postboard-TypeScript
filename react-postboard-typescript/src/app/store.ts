import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';

import signInPageReducer from '../components/SignInPage/signInPageSlice';
import postBoardPageReducer from '../components/PostBoardPage/postBoardPageSlice';

export const store = configureStore({
  reducer: {
    signInPage: signInPageReducer,
    postBoardPage: postBoardPageReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
