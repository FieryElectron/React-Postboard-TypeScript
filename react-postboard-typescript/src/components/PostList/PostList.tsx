import SinglePost from '../SinglePost/SinglePost';
import { useAppSelector } from '../../app/hooks';

import {
    selectPosts,
} from '../PostBoardPage/postBoardPageSlice';


const PostList = () => {
    const posts = useAppSelector(selectPosts);
    return (
        <div>
            {posts.map((post, index) => <SinglePost key={index} post={post}/>)}
        </div>
    )
}

export default PostList;