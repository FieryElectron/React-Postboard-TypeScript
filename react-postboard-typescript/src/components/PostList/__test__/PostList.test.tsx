import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';
import '@testing-library/jest-dom/extend-expect';

import PostList from '../PostList';
import { BrowserRouter as Router} from 'react-router-dom'
import { Provider } from 'react-redux';
import { store } from '../../../app/store';

let container: HTMLElement;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container.remove();
});

it('empty test', ()=>{

});

it('renders PostList without crashing', () => {
  act(() => {
    render(<Provider store={store}><Router><PostList /></Router></Provider>);
  });
});

it('matchs snapshot 1', ()=>{
  const tree = renderer.create(<Provider store={store}><Router><PostList/></Router></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});