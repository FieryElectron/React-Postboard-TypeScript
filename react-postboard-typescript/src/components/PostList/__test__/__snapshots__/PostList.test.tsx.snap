// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matchs snapshot 1 1`] = `<div />`;
